plasmapy/simulation/particletracker.py

```diff
@@ -1,203 +1,134 @@
 """
-Class representing a group of particles.
+Class representing a group of particles moving in a plasma's electric and
+magnetic fields.
 """
 
 import numpy as np
-import scipy.interpolate as interp
-from plasmapy.atomic import atomic
-from astropy import constants
 from astropy import units as u
+import numba
+import tqdm.auto
+import xarray
+import warnings
+
+from plasmapy import atomic, formulary
+from plasmapy.utils.decorators import check_units
+from plasmapy.utils import PhysicsError
+from plasmapy.atomic import particle_input, Particle
+
+PLOTTING = False
 
 __all__ = [
     "ParticleTracker",
+    "ParticleTrackerSolution",
 ]
 
-class ParticleTracker:
-    """
-    Object representing a species of particles: ions, electrons, or simply
-    a group of particles with a particular initial velocity distribution.
+@numba.njit(parallel = True)
+def _boris_push(x, v, b, e, hqmdt, dt):
```

Comment:
Perhaps de-abbreviate the variable name `hqmdt`?  Maybe something along the lines of `half_q_m_dt`, but there may be something better.

Reply:
I actually think I'll just change it to q, m and take the performance hit of dividing two floats with numba on each step... :D and then if it's a private variable it needs no long name but a comment: # half charge /m mass timestep.